### PR TITLE
Clean usage of deprecated products, fix nextChargeDate

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: payment-service/src/main/java/com/hedvig/paymentservice/services/bankAccounts/BankAccountServiceImpl.java [hunk 1]
```diff
@@ -45,20 +47,31 @@ public BankAccount getBankAccount(String memberId) {
...
+    YearMonth currentPeriod = YearMonth.of(today.getYear(), today.getMonth());
+    LocalDate chargeDateCurrentPeriod = getChargeDateOfPeriod(currentPeriod);
+    if (!today.isAfter(chargeDateCurrentPeriod)) {
+      return chargeDateCurrentPeriod;
     }
+    return getChargeDateOfPeriod(currentPeriod.plusMonths(1));
+  }
 
-    return LocalDate.of(YearMonth.now().getYear(), YearMonth.now().getMonth(), 27);
+  //TODO: Handle red days
```
Comment (FredrikAreschoug) [inline payment-service/src/main/java/com/hedvig/paymentservice/services/bankAccounts/BankAccountServiceImpl.java:65]: That is going to be fun! :smile:

**Review by FredrikAreschoug — APPROVED** [review]
Nice work!

File: payment-service/src/main/java/com/hedvig/paymentservice/services/bankAccounts/BankAccountServiceImpl.java [hunk 2]
```diff
@@ -45,20 +47,31 @@ public BankAccount getBankAccount(String memberId) {
...
+  static LocalDate getChargeDateOfPeriod(YearMonth period) {
+    LocalDate chargeDateThisMonth = period.atDay(27);
+    if (chargeDateThisMonth.getDayOfWeek() == DayOfWeek.SATURDAY) {
+      chargeDateThisMonth = chargeDateThisMonth.plusDays(2);
+    }
+    if (chargeDateThisMonth.getDayOfWeek() == DayOfWeek.SUNDAY) {
+      chargeDateThisMonth = chargeDateThisMonth.plusDays(1);
+    }
+    return chargeDateThisMonth;
```
Comment (meletios) [inline payment-service/src/main/java/com/hedvig/paymentservice/services/bankAccounts/BankAccountServiceImpl.java:74]: do we need to see if the member's contract is active? 

If its *not* active, the clients are expecting null

**Review by meletios — CHANGES_REQUESTED** [review]
One comment for the charging date logic

**Review by meletios — APPROVED** [review]

**Pull request merged**